lmcache/v1/lookup_client/lmcache_lookup_client.py

```diff
@@ -157,20 +200,30 @@ def lookup(
             ]
 
         results = []
+        failed_rank = -1
         try:
             for i in range(self.num_ranks):
+                failed_rank = i
                 self.sockets[i].send_multipart(msg_buf, copy=False)
 
             # TODO(Jiayi): we can use zmq poll to optimize a bit
             for i in range(self.num_ranks):
+                failed_rank = i
                 resp = self.sockets[i].recv()
                 result = int.from_bytes(resp, "big")
                 results.append(result)
-        except zmq.Again:
-            logger.error(f"Timeout occurred for rank {i}")
+        except zmq.Again as e:
+            logger.error(
+                f"Timeout occurred for rank {failed_rank}, "
+                f"recreating all sockets. Error: {str(e)}"
+            )
```

Comment:
nit: it's better not to use f-string when using loggers.

```suggestion
            logger.error(
                "Timeout occurred for rank %d, "
                "recreating all sockets. Error: %s",
                failed_rank, e
            )
```